### PR TITLE
add python 3.9, 3.10, 3.11 support, drop Travis CI configuration for GitHub CI

### PR DESCRIPTION
**Pull request by fmigneault**
## Description

Because some metadata was missing, multiple requirement validators fail to detect the Python 3 support, which flags this package in every other location that it gets used. While at it, I applied the most recent versions of Python 3 to ensure they are tested as well.

## Changes

* add missing classifiers and python requirements to ``setup.py`` to allow validators to detect appropriate versions
* add python 3.9, 3.10 and 3.11 to the supported versions in ``setup.py`` and validate them in Travis CI
* fix ``test_noaa`` with the target THREDDS server responding differently than originally tested

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ python:
   - "3.6"
   - "3.7"
   - "3.8"
+  - "3.9"
+  - "3.10"
+  - "3.11"
```
Comment (Zeitsperre) [inline .travis.yml:8]: Are there still projects around using Travis CI??

File: setup.py [hunk 2]
```diff
@@ -17,7 +17,15 @@
     'Operating System :: MacOS :: MacOS X',
     'Operating System :: Microsoft :: Windows',
     'Operating System :: POSIX',
```
Comment (Zeitsperre) [inline setup.py:19]:
```suggestion
```
This is probably not needed alongside `OS Independent`

File: .travis.yml (hunk 1 above)
Comment (fmigneault) [reply .travis.yml:8]: I don't know. I just assumed it was still using it. I wouldn't mind switching it to GitHub CI.

File: setup.py (hunk 2 above)
Comment (fmigneault) [reply setup.py:19]: Indeed, but it is more verbose to leave them IMO.

File: .travis.yml (hunk 1 above)
Comment (Zeitsperre) [reply .travis.yml:8]: I can write something up pretty quickly for GitHub CI in another PR if that interests you.

**Comment by fmigneault** [comment]
@cehbrecht  
FYI, I refer to eventual tag `0.4.4` in other PRs when this is ready. 
This is based on the latest version `0.4.3` from the repository.
However, this release is not available when targeting it (`0.4.2` is the latest).

File: .travis.yml (hunk 1 above)
Comment (fmigneault) [reply .travis.yml:8]: I can copy-paste a config from another repository as well. 
It is not a too complicated CI definition.

Comment (Zeitsperre) [reply .travis.yml:8]: I'll pass you something; I've been doing this all week, haha.

Comment (fmigneault) [reply .travis.yml]: Already done 😅

**Review by Zeitsperre — APPROVED** [review]

**Review by cehbrecht — APPROVED** [review]

**Comment by cehbrecht** [comment]
@fmigneault thanks 🙂

**Comment by cehbrecht** [comment]
> 



> This is based on the latest version `0.4.3` from the repository.
> However, this release is not available when targeting it (`0.4.2` is the latest).

@fmigneault I will upload 0.4.3 to pypi.

**Comment by fmigneault** [comment]
@cehbrecht  
Is it possible to have the mentioned `0.4.4` as well?

**Pull request merged**